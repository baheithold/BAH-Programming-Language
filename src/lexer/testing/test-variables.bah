# This test file tests various variable use cases

var x = 7;
x = 10;

var y = 10.6;
y = y + x;
y += 2;

var isOdd = false;
var isEven = true;
isOdd = !isEven;
isEven = !isOdd;

var nullify = null;
