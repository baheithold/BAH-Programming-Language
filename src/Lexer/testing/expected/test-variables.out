VAR_KEYWORD
ID_TYPE x
ASSIGN_BINARY
INTEGER_TYPE 7
SEMICOLON
ID_TYPE x
ASSIGN_BINARY
INTEGER_TYPE 10
SEMICOLON
VAR_KEYWORD
ID_TYPE y
ASSIGN_BINARY
REAL_TYPE 10.600000
SEMICOLON
ID_TYPE y
ASSIGN_BINARY
ID_TYPE y
PLUS_BINARY
ID_TYPE x
SEMICOLON
ID_TYPE y
PLUS_ASSIGN_BINARY
INTEGER_TYPE 2
SEMICOLON
VAR_KEYWORD
ID_TYPE isOdd
ASSIGN_BINARY
BOOLEAN_TYPE
SEMICOLON
VAR_KEYWORD
ID_TYPE isEven
ASSIGN_BINARY
BOOLEAN_TYPE
SEMICOLON
ID_TYPE isOdd
ASSIGN_BINARY
NEGATE_UNARY
ID_TYPE isEven
SEMICOLON
ID_TYPE isEven
ASSIGN_BINARY
NEGATE_UNARY
ID_TYPE isOdd
SEMICOLON
VAR_KEYWORD
ID_TYPE nullify
ASSIGN_BINARY
NULL_TYPE
SEMICOLON
END_OF_INPUT
